src/app/utils/utils.ts

```diff
@@ -1,7 +1,9 @@
+import firebase from 'firebase';
+
 // These are the datatypes supported by firestore. Arrays of arrays are not
 // supported, but arrays containing objects containing arrays are, which is what
 // is encoded in these types.
-export type JSONPrimitive = string | number | boolean | null;
+export type JSONPrimitive = firebase.firestore.FieldValue | string | number | boolean | null;
```

Comment:
Hmm ça semble un peu casser les couches d'abstraction / séparation des concern, non ?

Reply:
"// These are the datatypes supported by firestore."
C'est pas vraiment séparé de firestore comme fichier, de nature
Ceci dit, oui carrément, ce ticket casse un peu les couches!

Une idée ?

Reply:
Quid de distinguer entre
  - Un JSON firestorable: ça peut être utilisé partout, notamment pour encoder des move. C'est entièrement portable peu importe si on utilise firestore ou pas; donc si un jour faut se passer de firestore, tout ce code n'a pas besoin d'être adapté
  - Un JSON qui peut contenir des tags firestore en plus, tel qu'ici `FieldValue`. En gros un JSON avec des trous que firestore viendra compléter. Utile pour tout ce qui est logique de DB, pas utilisé dans les jeux/move.
Le second peut être défini à partir du premier:
`export type JSONPrimitiveWithFieldValue = JSONPrimitive | FieldValue`
(et faut cascader ça aux autres JSON*)

Reply:
ça me semble ok, je tickette ça.